src/codeyCommand.ts

```diff
@@ -10,72 +33,256 @@ export type SapphireMessageExecuteType = (
+      commandOption.setDescription(option.description);
+      commandOption.setRequired(option.required);
+      return <SlashCommandBuilder>builder.addNumberOption(commandOption);
+    case CodeyCommandOptionType.ATTACHMENT:
+      commandOption = new SlashCommandAttachmentOption();
+      commandOption.setName(option.name);
+      commandOption.setDescription(option.description);
+      commandOption.setRequired(option.required);
+      return <SlashCommandBuilder>builder.addAttachmentOption(commandOption);
+    default:
+      throw new Error(`Unknown option type.`);
+  }
+};
+
+/** The possible types of the value of a command option */
+export type CodeyCommandArgumentValueType = string | number | boolean | undefined;
+/** A standardized dictionary that stores the arguments of the command */
+export type CodeyCommandArguments = { [key: string]: CodeyCommandArgumentValueType };
+
+/** Details for the codey command (or subcommand) */
+export class CodeyCommandDetails {
+  /** The name of the command */
+  name!: string;
+  /** The aliases of the command (for regular commands) */
+  aliases: string[] = [];
+  /** A short description of the command (shown in the slash command menu) */
+  description = `Codey command for ${this.name}`;
+  /** A longer description of the command (shown in the help menu for the command) */
+  detailedDescription: string = this.description;
+
+  /** The message to display when the command is executing (for slash commands) */
   messageWhenExecutingCommand!: string;
-  // The function to be called when the command is executing
+  /** The function to be called to execute the command */
   executeCommand!: SapphireMessageExecuteType;
-  // The message to display if the command fails
+  /** The message to display if the command fails */
   messageIfFailure = 'Codey backend error - contact a mod for assistance';
-
+  /** A flag to indicate if the command response is ephemeral (ie visible to others) */
   isCommandResponseEphemeral = true;
-  // Type of response Codey command sends
+  /** Type of response the Codey command sends */
   codeyCommandResponseType: CodeyCommandResponseType = CodeyCommandResponseType.STRING;
 
-  commandOptions!: CodeyCommandOptions;
+  /** Options for the Codey command */
+  options: CodeyCommandOption[] = [];
+  /** Subcommands under the CodeyCommand */
+  subcommandDetails: { [name: string]: CodeyCommandDetails } = {};
+}
+
+/** Sets the command subcommand in the slash command builder */
+const setCommandSubcommand = (
+  builder: SlashCommandBuilder,
+  subcommandDetails: CodeyCommandDetails
+): SlashCommandSubcommandsOnlyBuilder => {
+  const subcommandBuilder = new SlashCommandSubcommandBuilder();
+  subcommandBuilder.setName(subcommandDetails.name);
+  subcommandBuilder.setDescription(subcommandDetails.description);
+  for (const commandOption of subcommandDetails.options) {
+    setCommandOption(subcommandBuilder, commandOption);
+  }
+  return builder.addSubcommand(subcommandBuilder);
```

Comment:
what does making a subcommand do? I see you're still manually checking for subcommands later on in the code.